### ISSUE TITLE
Feature Request: Azure High Availability Ports

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-azurerm#372]
_This issue was originally opened by @robrankin as hashicorp/terraform#16215. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Microsoft has just announced the Preview availability of a "High Availability Ports" for the Azure Load Balancer:

https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-ha-ports-overview

It's a pretty great addition to the LB, would be extremely useful to have it exposed in Terraform.

**Comment by justaugustus** [comment]
@metacpp pinging you on this one, as you assigned yourself to #579.

**Comment by mb290** [comment]
Is there any chance this feature is scoped for implementation? It's the last piece of my vDC Terraform implementation and saves me creating hundreds of rules on my load balancer in front of next-gen firewalls.

**Comment by katbyte** [comment]
Hi @justaugustus,

Thanks for the issue, but we are unfortunately still waiting on the GO SDK for Azure to support this.

**Comment by EtienneDeneuve** [comment]
From our side, we have successfully deployed a Azure Load Balancer with HA Ports with the following configuration :
```hcl
resource "azurerm_lb_rule" "lb_haports_rule" {
  backend_port = 0
  frontend_ip_configuration_name = "${azurerm_lb.private-lb-gateway.frontend_ip_configuration.0.name}"
  frontend_port = 0
  loadbalancer_id = "${azurerm_lb.private-lb-gateway.id}"
  name = "lb_haports_rule"
  protocol = "All"
  backend_address_pool_id = "${azurerm_lb_backend_address_pool.gateway_backend_pool.id}"
  probe_id = "${azurerm_lb_probe.lb_gateway_probe_fw.id}"
  resource_group_name          = "${var.resource_group_name}"
}
```

And it's working ! We simply add "All" in protocol.

**Comment by asubmani** [comment]
@EtienneDeneuve .. Terraform 0.11.8 throws an error when I run the workaround config. The error is 
`"backend_port is not a valid port number: 'x00'`
My config looks like:
```hcl
resource "azurerm_lb_rule" "lb_haports_rule" {
  protocol = "All"
  backend_port = 0
  frontend_ip_configuration_name = "${azurerm_lb.ilbmain.frontend_ip_configuration.0.name}"
  frontend_port = 0
  loadbalancer_id = "${azurerm_lb.ilbmain.id}"
  name = "lb_haports_rule"  
  backend_address_pool_id = "${azurerm_lb_backend_address_pool.ILB2PaloAlto.id}"
  probe_id = "${azurerm_lb_probe.paloalto_probe.id}"
  resource_group_name  = "${data.azurerm_resource_group.pa-rg.name}"
}
```

**Comment by LW81** [comment]
Hi,
just bumped into the same issue and also for me the workaround is no longer working.

What surprises me is that the referenced issue for the go-sdk is also closed !
Can somebody please have a look to see how this can be done.

**Comment by DYNSOL** [comment]
I'm also getting 

> "backend_port is not a valid port number: 'x00'

Yet if you leave it to "All" and put in a port it moans that it must be set to 0, which then errors.

> Message="A rule that supports any port and protocol load balancing must have the following settings: FrontendPort: 0, BackendPort: 0, and Protocol: Protocol.All.

**Comment by katbyte** [comment]
@DYNSOL, @LW81, @asubmani 

Additional validation was added to this resource breaking the workaround. I have opened #1951 to allow `0` for the ports again.

**Comment by LW81** [comment]
Excellent ! Many thx for taking care !

kt <notifications@github.com> schrieb am Do., 20. Sep. 2018, 00:08:

**Comment by tombuildsstuff** [comment]
Fixed via #1951
